// @e-group/react-material-components/dist/Button/Button.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import withStyles from '@material-ui/core/styles/withStyles';
import MUIButton from '@material-ui/core/Button/Button';
import { CircularProgress } from '@material-ui/core';
import styles from './styles';
export var ButtonComponent = function ButtonComponent(_ref) {
  var classes = _ref.classes,
      classNameProp = _ref.className,
      children = _ref.children,
      loading = _ref.loading,
      success = _ref.success,
      MUIButtonProps = _ref.MUIButtonProps,
      CircularProgressProps = _ref.CircularProgressProps,
      other = _objectWithoutProperties(_ref, ["classes", "className", "children", "loading", "success", "MUIButtonProps", "CircularProgressProps"]);

  var _ref2 = MUIButtonProps || {},
      MUIButtonClassNameProp = _ref2.className,
      fullWidth = _ref2.fullWidth,
      otherMUIButtonProps = _objectWithoutProperties(_ref2, ["className", "fullWidth"]);

  var _ref3 = CircularProgressProps || {},
      CircularProgressNameProp = _ref3.className,
      otherCircularProgressProps = _objectWithoutProperties(_ref3, ["className"]);

  return React.createElement("div", _extends({
    className: classNames(classes.root, _defineProperty({}, classes.fullWidth, fullWidth), classNameProp)
  }, other), React.createElement(MUIButton, _extends({
    className: classNames(_defineProperty({}, classes.success, success), MUIButtonClassNameProp),
    disabled: loading,
    fullWidth: fullWidth
  }, otherMUIButtonProps, {
    children: children
  })), loading && React.createElement(CircularProgress, _extends({
    size: 24,
    className: classNames(classes.progress, CircularProgressNameProp)
  }, otherCircularProgressProps)));
};
ButtonComponent.propTypes = {
  /**
   * Override or extend the styles applied to the component.
   */
  classes: PropTypes.object.isRequired,

  /**
   * JSX attribute
   */
  className: PropTypes.string,

  /**
   * The content of the component.
   */
  children: PropTypes.node,

  /**
   * The button's loading status
   */
  loading: PropTypes.bool,

  /**
   * The button's success status
   */
  success: PropTypes.bool,

  /**
   * MUI Button Props
   */
  MUIButtonProps: PropTypes.object,

  /**
   * Circular Progress Props
   */
  CircularProgressProps: PropTypes.object
};
ButtonComponent.defaultProps = {
  loading: false,
  success: false
};
var Button = withStyles(styles)(ButtonComponent);
Button.displayName = 'Button';
export default Button;
