// @e-group/react-material-components/dist/Loader/Loader.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import componentPropType from '@material-ui/utils/componentPropType';
import withStyles from '@material-ui/core/styles/withStyles';
import CircularProgress from '@material-ui/core/CircularProgress';
import styles from './styles';
export var LoaderComponent = function LoaderComponent(_ref) {
  var classes = _ref.classes,
      classNameProp = _ref.className,
      Loader = _ref.loader,
      LoaderProps = _ref.LoaderProps,
      align = _ref.align,
      other = _objectWithoutProperties(_ref, ["classes", "className", "loader", "LoaderProps", "align"]);

  var className = classNames(classes.root, classes[align], classNameProp);
  return React.createElement("div", _extends({
    className: className
  }, other), React.createElement(Loader, LoaderProps));
};
LoaderComponent.propTypes = {
  /**
   * Override or extend the styles applied to the component.
   */
  classes: PropTypes.object.isRequired,

  /**
   * JSX attribute
   */
  className: PropTypes.string,

  /**
   * The loader component.
   * Either a string to use a DOM element or a component.
   */
  loader: componentPropType,

  /**
   * Default loader component is MUI's CircularProgress component
   * Please read [official doc](https://material-ui.com/api/circular-progress/)
   */
  LoaderProps: PropTypes.object,

  /**
   * Defined the loader position.
   * enum: 'stretch' | 'center'
   */
  align: PropTypes.string
};
LoaderComponent.defaultProps = {
  loader: CircularProgress,
  align: 'stretch'
};
var Loader = withStyles(styles)(LoaderComponent);
Loader.displayName = 'Loader';
export default Loader;
