// @e-group/react-material-components/dist/Container/Container.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import componentPropType from '@material-ui/utils/componentPropType';
import withStyles from '@material-ui/core/styles/withStyles';
import styles from './styles';
export var ContainerComponent = function ContainerComponent(_ref) {
  var classes = _ref.classes,
      classNameProp = _ref.className,
      Component = _ref.component,
      maxWidth = _ref.maxWidth,
      other = _objectWithoutProperties(_ref, ["classes", "className", "component", "maxWidth"]);

  var className = classNames(classes.root, _defineProperty({}, classes[maxWidth], maxWidth !== false), classNameProp);
  return React.createElement(Component, _extends({
    className: className
  }, other));
};
ContainerComponent.propTypes = {
  /**
   * Override or extend the styles applied to the component.
   */
  classes: PropTypes.object.isRequired,

  /**
   * JSX attribute
   */
  className: PropTypes.string,

  /**
   * The content of the component.
   */
  children: PropTypes.node,

  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: componentPropType,

  /**
   * Determine the max-width of the container.
   * The container width grows with the size of the screen.
   * Set to `false` to disable `maxWidth`.
   */
  maxWidth: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', false])
};
ContainerComponent.defaultProps = {
  component: 'div',
  maxWidth: 'md'
};
var Container = withStyles(styles)(ContainerComponent);
Container.displayName = 'Container';
export default Container;
